### PR TITLE
sql: do not use the unexported getTimeSrv function

### PR DESCRIPTION
**Pull request by gabor**
the shared sql functionality codebase in grafana uses the following import:
```ts
import { getTimeSrv } from 'app/features/dashboard/services/TimeSrv';
```

as working on making the core sql datasources external-ish, we cannot import unexported code from grafana.

this PR uses a workaround where we use the exported `getTemplateSrv()` to get the time range using `getTemplateSrv().replace('$__from $__to')`.

long term it would be better to completely avoid relying on a singleton-based get-time-range approach, because there may be places where there are multiple time-ranges "current" at the same time. i investigated this approach here, but it is not easy:
1. we would need to migrate from `.metricFindQuery()` to `this.variables`
2. the API for `.runSql()` would have to get a time-range param, so the callers would have to calculate it "somehow".

how to test:
1. open devtools in the browser, we will look at the `query` ajax requests
2. create a dashboard query variable for one of the core sql datasources, for example postgres, you can use a "dummy" sql query like "SELECT 42". save the query variable.
3. go back to the dashboard, reload the page
4. check the ajax requests. find the one doing the SELECT 42. verify that the "from" and "to" json attributes in the request-body match the currently selected time range in the dashboard.

### CODE REVIEW

**Review by aangelisc — APPROVED** [review]

**Review by aangelisc — APPROVED** [review]

File: public/app/features/plugins/sql/utils/time.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import { dateTime, rangeUtil, TimeRange } from '@grafana/data';
+import { TemplateSrv } from '@grafana/runtime';
+
+const RANGE_RE = /^(\d+) (\d+)$/; // find digits, then a space, then digits again.
+
+// there is no exported "getTimeSrv" functionality,
+// so we use the template server to access the current time range.
+export function getCurrentTimeRange(templateSrv: TemplateSrv): TimeRange {
+  const text = templateSrv.replace('$__from $__to');
```
Comment (ifrost) [inline public/app/features/plugins/sql/utils/time.ts:9]: Explore updates TemplateSrv range only when a query is run, so `getCurrentTimeRange` may not get correct values, especially in split mode (though there are similar problem with TimeSrv). I was hoping we could correctly populate range in https://github.com/grafana/grafana/blob/main/public/app/features/query/components/QueryEditorRow.tsx#L294 so editors always get the correct time range as a prop.

**Comment by zoltanbedi** [comment]
Before we merge this I would be curious about other possibilities that you mentioned during our daily @gabor

File: public/app/features/plugins/sql/utils/time.ts (hunk 1 above)
Comment (gabor) [reply public/app/features/plugins/sql/utils/time.ts:9]: yeah, the situation is not perfect. all i can promise, is that this change will not make it more incorrect than before 😄 
(FYI:i'm looking into an alternative approach, see draft PR at https://github.com/grafana/grafana/pull/74800)

**Comment by gabor** [comment]
we'll go with a different approach (see https://github.com/grafana/grafana/pull/74800), closing this one.

**Pull request closed without merging**